[Affliction] Dreadful Calling trait

sMteX opened the pull request:
![image](https://user-images.githubusercontent.com/5068584/48925769-c4a74800-eec7-11e8-8876-fdbcd632512b.png)

Part of #2724. Added a helper function that should *estimate* the bonus part of Azerite traits that affect damage of spells (related issue: #2088). 

Assuming following:
- Azerite traits of this type affect *base* damage of spells
- For casters, this damage is determined by Intellect (and thus Spell Power) and the spell's SP coefficient
- so if `damage = (base + trait bonus) * coefficients`, that means both base and the trait bonus is scaled by same coefficients (like mastery, versatility etc.)
- and even if both damage parts gets scaled by the coefficients, their ratio stays the same regardless of it

We should be able to at least estimate the damage bonus with the following:
```
trait bonus = value that affects the damage
base damage = SP coefficient * current intellect
portion = trait bonus / (base damage + trait bonus)
bonus damage = real damage * portion
```
Example for Unstable Affliction and Dreadful Calling trait (assuming 100 bonus damage from trait, 6000 intellect, 0.16 SP coefficient and 4000 damage from event):
```
traitBonus = 100
intellect = 6000
SPcoeff = 0.16
actualDamage = 4000

baseDamage = 0.16 * 6000 = 960
portion = 100 / (960 + 100) = 0.094339
bonusDamage = 0.094339 * 4000 = 377.356
```

My source from which I got the idea for this, thinks this should be applicable to all Azerite traits (at least the Warlock ones), so I'd like to merge this before updating #2738 in order to not duplicate code in it as well. If others think this should apply to other specs/classes, it might be good idea to move it somewhere else into `common` probably.

Gebuz commented:
Yep, that's how all azerite traits work, so please do add the helper function in to /parser/core. 

Racials like Blood Fury still give AP/SP instead of primary stat iirc? Might need to use that instead of int.

For this trait, I think it should be made more obvious that the CDR is not included in the DPS estimate, and maybe rewrite the line about the int to put emphasis on it only being as accurate as our stat tracker (I still think we are missing a bunch of trinkets?)

sMteX commented:
I actually used the formula that was in Feeding Frenzy (Beast Mastery Hunter) and Wild Fleshrending (Feral Druid), it seemed more elegant. Function was moved to `parser/core/calculateBonusAzeriteDamage.js` and documented. I also refactored the said 2 modules to use this new function as well (tested on https://www.warcraftlogs.com/reports/TbA6KcXYCLtZd4D1/#fight=2&source=2 and https://www.warcraftlogs.com/reports/bCPHf81hrnyFZp2K/#fight=3&source=4, nothing changed in the output).

Proof that the 2 (my original and the new formula) are the same:
Assuming `total = (base + bonus) * modifiers`
Original:
 ```
ratio = (bonus * modifiers) / ((base + bonus) * modifiers) = bonus / (base + bonus)
Result = total * ratio = (total * bonus) / (base + bonus)
```
New: 
```
modifiers = total / (base + bonus)
Result = bonus * modifiers = (bonus * total) / (base + bonus)
```
Also, updated tooltip for Dreadful Calling
![image](https://user-images.githubusercontent.com/5068584/48944451-ff8c9880-ef26-11e8-8ca4-15c0365c709a.png)
